### DISCUSSION TITLE
Vehicle Routing PRoblem - with many "consolidation hubs"

### DISCUSSION THREAD

**Discussion by popandre** [TimefoldAI/timefold-solver#1218] [General]
Some time ago, I delved deeply into the Vehicle Routing Problem and was able to create a relatively complex solution based on OptaPlanner. Various challenges (data) and constraints were taken into account, and I felt I was on the right track. I had seen good and understandable results (transport plans), BUT I failed at the following task:

There are many consolidation points (hubs?) in the network where, for example, many smaller vehicles drop off their loads, which are then taken to a destination by one  large truck(s) (everything is many-to-many). It is equally possible that some of the smaller vehicles would be better off driving directly to one of the destinations, while for others it makes sense to drop off the load at the hub.

I was never able to solve this problem satisfactorily. I had the impression that the solution space explodes when these hubs come into play.

Question: Has such a scenario ever been solved with timefold and successfully deployed productively in a professional environment?

**Comment by ge0ffrey** [comment]
I call this the proxy-depot design patten.

Presume you have bicycle A, B, C. They all start from a proxy-depot, which isn't a real depot.
That proxy-depot is the drop off point of the large truck. The large truck only drives from it's depot to the proxy-depot (a single visit) **if and only if** one of the bicycles drops something off.

That's your model.

Only penalize the cost for the truck if and only if there is at least one visit assigned to at least one of the bicycles starting from the proxy-depot.

**Reply by popandre** [reply]
Thanks a lot for explanation!
